### PR TITLE
UI tests

### PR DESCRIPTION
**Pull request by mik629**
* this is an attempt to learn UI testing

### CODE REVIEW

File: app/src/androidTest/java/com/github/mik629/android/fundamentals/ui/movieslist/MoviesListTest.kt [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+class MoviesListTest : TestCase() {
+    @Test
+    fun testRecyclerViewItemLayout() {
+        ActivityScenario.launch(AppActivity::class.java)
+        MoviesListScreen {
+            movieList {
+                firstChild<MoviesListScreen.MovieItem> {
+                    moviePoster {
+                        isVisible()
```
Comment (mik629) [inline app/src/androidTest/java/com/github/mik629/android/fundamentals/ui/movieslist/MoviesListTest.kt:18]: I guess I need to mock Repository with test data to be able to check the content (if I need to check content at all).
I also wonder to what level should I test. Like every element's style, margins, color, etc. or it would be better to test at higher level: like what happens with UI/app when some part isn't working (e.g. errors from api: 400, 404, 500, no network, etc, some corner cases for input (though in my app, there is no input yet)  data maybe) 🤔
Probably, I should read some book on UI testing.

Comment (rHlushan) [reply app/src/androidTest/java/com/github/mik629/android/fundamentals/ui/movieslist/MoviesListTest.kt:18]: first of all: just start from unit tests
next: test primary functionality first
try to just write empty functions with names, describing what case u want test
